service/worker/scheduler/responsebuilder_test.go

```diff
@@ -0,0 +1,328 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package scheduler
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	failurepb "go.temporal.io/api/failure/v1"
+	historypb "go.temporal.io/api/history/v1"
+	"google.golang.org/protobuf/proto"
+
+	schedspb "go.temporal.io/server/api/schedule/v1"
+	"go.temporal.io/server/common/log"
+)
+
+func TestResponseBuilder(t *testing.T) {
+	nilLogger := log.NewNoopLogger()
+	request := schedspb.WatchWorkflowRequest{
+		Execution: &common.WorkflowExecution{WorkflowId: "workflow-id-1"},
+	}
+	t.Run("when execution status is RUNNING", func(t *testing.T) {
+		status := enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING
+		t.Run("when LogPoll requested will return errTryAgain", func(t *testing.T) {
+			longPollRequest := schedspb.WatchWorkflowRequest{LongPoll: true}
+			rb := newResponseBuilder(&longPollRequest, status, nilLogger, eventStorageSize-recordOverheadSize)
+			event := historypb.HistoryEvent{}
+
+			response, err := rb.Build(&event)
+
+			assertError(t, err, errTryAgain)
+			assertResponseIsNil(t, response)
+		})
+		t.Run("when it is not a LongPoll will return response with nil result", func(t *testing.T) {
+			rb := newResponseBuilder(&request, status, nilLogger, eventStorageSize-recordOverheadSize)
+			event := historypb.HistoryEvent{}
+
+			response, err := rb.Build(&event)
+
+			assertError(t, err, nil)
+			assertResponseResult(t, response, nil)
+			assertResponseStatus(t, response, enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING)
+
+		})
+	})
+
+	t.Run("when status is COMPLETED", func(t *testing.T) {
+		status := enumspb.WORKFLOW_EXECUTION_STATUS_COMPLETED
+		t.Run("when no attrs return errNoAttrs", func(t *testing.T) {
+			rb := newResponseBuilder(&request, status, nilLogger, eventStorageSize-recordOverheadSize)
+			event := historypb.HistoryEvent{}
+
+			response, err := rb.Build(&event)
+
+			assertError(t, err, errNoAttrs)
+			assertResponseIsNil(t, response)
+		})
+		t.Run("when NewExecutionRunId is non empty returns errFollow", func(t *testing.T) {
+			rb := newResponseBuilder(&request, status, nilLogger, eventStorageSize-recordOverheadSize)
```

Comment:
`rb` can be moved to the top of this sub test since it's the same in all nested test cases.